### PR TITLE
zh-trans: /docs/setup/production-environment/container-runtimes.md

### PR DESCRIPTION
**Pull request by yuxiaobo96**
Signed-off-by: yuxiaobo <yuxiaobogo@163.com>

### CODE REVIEW

**Comment by yuxiaobo96** [comment]
@tengqm @markthink 
Is this my problem or the problem of deploying the test?
I have read the error message and it looks like this:
`10:25:17 AM: [ERROR] Deploy logs are currently unavailable. We are working on resolving the issue.`
Please help me !

**Comment by tengqm** [comment]
Tried close and reopen, let's see if the problem is solved.

**Comment by tengqm** [comment]
@yuxiaobo96 If you use a proxy, you will see the error output now. The error is about the `tabs` shortcode.

**Comment by tanjunchen** [comment]
@yuxiaobo96 
image

**Comment by yuxiaobo96** [comment]
Thanks for your advice.

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+{{< caution >}}
+<!--
+A flaw was found in the way runc handled system file descriptors when running containers.
+A malicious container could use this flaw to overwrite contents of the runc binary and
+consequently run arbitrary commands on the container host system.
+
+Please refer to this link for more information about this issue
+[cve-2019-5736 : runc vulnerability ] (https://access.redhat.com/security/cve/cve-2019-5736)
+-->
+当运行容器，runc 处理系统文件描述符时，发现一个漏洞。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
我们发现 runc 在运行容器，处理系统文件描述符时存在一个漏洞。
```
翻译过来的词句还是要符合表达、阅读习惯。

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+### Cgroup 驱动程序
+
+<!--
+When systemd is chosen as the init system for a Linux distribution, the init process generates
+and consumes a root control group (`cgroup`) and acts as a cgroup manager. Systemd has a tight
+integration with cgroups and will allocate cgroups per process. It's possible to configure your
+container runtime and the kubelet to use `cgroupfs`. Using `cgroupfs` alongside systemd means
+that there will then be two different cgroup managers.
+-->
+当选择一个 Linux 系统发行版作为初始化系统时，初始化进程会生成一个 root 控制组 （`cgroup`），并使用它充当 cgroup 管理器。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
当某个 Linux 系统发行版使用 systemd 作为其初始化系统时，初始化进程会生成并使用一个 root 控制组 （`cgroup`），并充当 cgroup 管理器。
```

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 3]
```diff
@@ -0,0 +1,584 @@
...
+系统具有紧密的 cgroup 集成，并将为每个进程分配 cgroup。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
systemd 与 cgroup 集成紧密，并将为每个进程分配 cgroup。
```

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
+您可以配置容器运行时和 kubelet 去使用 `cgroupf`。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
您也可以配置容器运行时和 kubelet 使用 `cgroupfs`。
```

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 5]
```diff
@@ -0,0 +1,584 @@
...
+连同系统一起使用 `cgroupf` 意味着将有两个不同的 cgroup 管理器。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
连同 systemd 一起使用 `cgroupfs` 意味着将有两个不同的 cgroup 管理器。
```

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 6]
```diff
@@ -0,0 +1,584 @@
...
+A single cgroup manager will simplify the view of what resources are being allocated
+and will by default have a more consistent view of the available and in-use resources. When we have
+two managers we end up with two views of those resources. We have seen cases in the field
+where nodes that are configured to use `cgroupfs` for the kubelet and Docker, and `systemd`
+for the rest of the processes running on the node becomes unstable under resource pressure.
+-->
+控制组用来约束分配给进程的资源。
+单个 cgroup 管理器将简化分配资源的视图，并且默认情况下将对可用资源和使用中的资源具有更一致的视图。
+当有两个管理器时，最终将对这些资源产生两种视图。
+在此领域我们已经看到案例，其中为 kubelet 和 docker 配置节点来使用 `cgroupf`，在节点上运行的其余进程在资源压力下会变得不稳定。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
在此领域我们已经看到案例，某些节点配置为让 kubelet 和 docker 使用 `cgroupfs`，而节点上运行的其余进程则使用 systemd；这类节点在资源压力下会变得不稳定。
```

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 7]
```diff
@@ -0,0 +1,584 @@
...
+
+<!--
+Changing the settings such that your container runtime and kubelet use `systemd` as the cgroup driver
+stabilized the system. Please note the `native.cgroupdriver=systemd` option in the Docker setup below.
+-->
+更改设置，以便您的容器运行时和 kubelet 使用 `systemd` 作为 cgroup 驱动来稳定系统。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
更改设置，令容器运行时和 kubelet 使用 `systemd` 作为 cgroup 驱动，以此使系统更为稳定。
```

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 8]
```diff
@@ -0,0 +1,584 @@
...
+{{< caution >}}
+<!--
+Changing the cgroup driver of a Node that has joined a cluster is highly unrecommended.
+If the kubelet has created Pods using the semantics of one cgroup driver, changing the container
+runtime to another cgroup driver can cause errors when trying to re-create the PodSandbox
+for such existing Pods. Restarting the kubelet may not solve such errors. The recommendation
+is to drain the Node from its workloads, remove it from the cluster and re-join it.
+-->
+强烈建议不要更改已加入集群的节点的 cgroup 驱动。
+如果 kubelet 已经使用 cgroup 驱动的语义创建了 pod，则当尝试为现有的 pod 重新创建 PodSandbox 时，其他的 cgroup 驱动程序的运行时可能会产生错误。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
如果 kubelet 已经使用某 cgroup 驱动的语义创建了 pod，尝试更改运行时以使用别的 cgroup 驱动，为现有 Pods 重新创建 PodSandbox 时会产生错误。
```

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 9]
```diff
@@ -0,0 +1,584 @@
...
+重启 kubelet 或许不能解决此类问题。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
重启 kubelet 也可能无法解决此类问题。
```

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 10]
```diff
@@ -0,0 +1,584 @@
...
+推荐从工作负载中排出节点，将其从集群中删除并重新加入。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
推荐将工作负载逐出节点，之后将节点从集群中删除并重新加入。
```

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 11]
```diff
@@ -0,0 +1,584 @@
...
+Version 18.06.2 is recommended, but 1.11, 1.12, 1.13, 17.03 and 18.09 are known to work as well.
+Keep track of the latest verified Docker version in the Kubernetes release notes.
+
+Use the following commands to install Docker on your system:
+-->
+## Docker
+
+在您的每台机器上安装 Docker。
+推荐安装 18.06.2 版本，但是 1.11、1.12、1.13、17.03 和 18.09 版本也是可以的。
+在 Kubernetes 发行说明中保持跟随经过验证的 Docker 最新版本。
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
请跟踪 Kubernetes 发行说明中经过验证的 Docker 最新版本变化。
```

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 12]
```diff
@@ -0,0 +1,584 @@
...
+apt-get update
+apt-get install software-properties-common
+
+add-apt-repository ppa:projectatomic/ppa
+apt-get update
+
+# Install CRI-O
+apt-get install cri-o-1.15
+-->
+# 安装 prerequisites
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
# 安装必备软件
```

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 13]
```diff
@@ -0,0 +1,584 @@
...
+{{< tab name="CentOS/RHEL 7.4+" codelang="bash" >}}
+
+<!--
+# Install prerequisites
+yum-config-manager --add-repo=https://cbs.centos.org/repos/paas7-crio-115-release/x86_64/os/
+
+# Install CRI-O
+yum install --nogpgcheck cri-o
+-->
+# 安装 prerequisites
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]: 同上

File: content/zh/docs/setup/production-environment/container-runtimes.md [hunk 14]
```diff
@@ -0,0 +1,584 @@
...
+sysctl --system
+```
+-->
+## 容器
+
+本节包含使用 `containerd` 作为 CRI 运行时的必要步骤。
+
+使用以下命令在系统上安装容器：
+
+### Prerequisites
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/container-runtimes.md]:
```suggestion
### 准备环境
```

File: content/zh/docs/setup/production-environment/container-runtimes.md (hunk 1 above)
Comment (yuxiaobo96) [reply content/zh/docs/setup/production-environment/container-runtimes.md]: 好的，以后注意

**Comment by yuxiaobo96** [comment]
/assign @tengqm 
已修改

**Pull request merged**